### PR TITLE
Security ldap branch

### PR DESCRIPTION
**Pull request by irinaponomarevahaulmont**
First draft ready, please review.

### CODE REVIEW

File: content/modules/studio/pages/ldap.adoc [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+
+The *_Anonymous access_* checkbox should be left checked for the purpose of this guide, but if we need to connect to another server, the checkbox can be unchecked. In this case, *_Manager DN_* and *_Manager password_* should be filled in.
+
+For *_Authentication_* use `Bind`.
+
+*_User DN pattern_* is the directory, where the system will look for users. Enter `uid=\{0},ou=users`.
+
+*_Authorities_*: choosing how the authority users will be gained. For the purpose of this guide choose `Group member`.
+
+*_Group search base_* should be `ou=groups`.
```
Comment (alexbudarov) [inline content/modules/studio/pages/ldap.adoc:248]: add explanation:

> Base directory to search for authorities. The application will search for groups of which the user is a member (typically these will be groupOfNames or groupOfUniqueNames entries in the directory). Roles will be assigned to the user by searching under the "Group search base" with the default filter (member=<user's-DN>). The role name will be taken from the “ou” attribute of each match. 

(took from https://docs.spring.io/spring-security/site/docs/3.1.x/reference/ldap.html)

File: content/modules/studio/pages/ldap.adoc [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+image::security-ldap/dn-editor-2.png[align=center, width=700]
+
+However, for the purpose of this guide we need only one user in each group, so we don't need to carry out this step.
+
+Now let us create a second role. The steps will be the same as described above for the `ADMIN` role, but the role name will be `USER`. The user we pick for this role is `hlee`.
+
+Configuring the structure of our LDAP folder has been completed.
+
+[[connection-parameters]]
+=== Checking Connection Parameters.
```
Comment (alexbudarov) [inline content/modules/studio/pages/ldap.adoc:185]: remove dot in the end

**Review by alexbudarov — CHANGES_REQUESTED** [review]

File: content/modules/studio/pages/ldap.adoc [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+
+The following lines will be added to the `import` section of the class:
+
+[source, java]
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.ldap.userdetails.InetOrgPerson;
+
+Use *Ctrl+Space* (*⌘+Space* for macOS) to add these imports.
+
+With the modifications we have made, if the current user is an LDAP user (`InetOrgPerson`) and it has the `cn` field filled in, we retrieve its value and use it as the user's full name.
```
[the remark this answers is not in the document]
Comment (irinaponomarevahaulmont) [reply content/modules/studio/pages/ldap.adoc:455]: No, because LDAP sounds "el-di-ei-pi" when pronounced.

**Comment by irinaponomarevahaulmont** [comment]
Please review the first round of requested changes.

File: content/modules/studio/pages/ldap.adoc [hunk 4]
```diff
@@ -0,0 +1,471 @@
+= Security -- Using LDAP with Amplicode-based Projects
```
[the remark this answers is not in the document]
Comment (alexbudarov) [reply content/modules/studio/pages/ldap.adoc]: not fixed

[the remark this answers is not in the document]
Comment (irinaponomarevahaulmont) [reply content/modules/studio/pages/ldap.adoc]: OK, guess I misunderstood you.

**Comment by irinaponomarevahaulmont** [comment]
Please review.

**Pull request merged**